app/controllers/single_idp_journey_controller.rb

```
@@ -37,16 +37,24 @@ def continue_to_your_idp
   end
 
   def continue
-    select_viewable_idp_for_single_idp_journey(params.fetch('entity_id')) do |decorated_idp|
-      select_idp(decorated_idp.entity_id, decorated_idp.display_name)
-      redirect_to redirect_to_single_idp_path
+    if params_are_missing(%w(entity_id))
+      redirect_to start_path
+    else
+      select_viewable_idp_for_single_idp_journey(params.fetch('entity_id')) do |decorated_idp|
+        select_idp(decorated_idp.entity_id, decorated_idp.display_name)
+        redirect_to redirect_to_single_idp_path
+      end
     end
   end
 
   def continue_ajax
-    select_viewable_idp_for_single_idp_journey(params.fetch('entityId')) do |decorated_idp|
-      select_idp(decorated_idp.entity_id, decorated_idp.display_name)
-      ajax_idp_redirection_single_idp_journey_request(single_idp_cookie&.fetch('uuid', nil))
+    if params_are_missing(%w(entityId))
+      redirect_to start_path
```

this is an ajax call (`PUT`), so the `redirect_to` will not do anything, unless it returns an error code. We should return an error HTTP response so it falls back to the standard `POST` request and can do the redirection. Alternatively we don't need the check the params here is it will error out on the `params.fetch` and fallback anyway